### ISSUE TITLE
Tailwind styles not overwriting default <Text> styles

### ISSUE THREAD

**Issue by mckelveygreg** [resend/react-email#517]
Currently, the inline styles default to the `Text` component take precedence over tailwind styles that I'm passing to it. 

My workaround
```tsx
<Text
  style={{
  margin : undefined,
  fontSize: undefined,
  lineHeight: undefined,
 }}
 className="m-0 text-sm"
>
```

**Comment by mckelveygreg** [comment]
I did some more investigating because I noticed that there were tests for this situation. 

I looks like there may be 2 issues:
- when a div has a responsive tailwind class (ex. `sm:w-1`), all classes get hoisted to the `<head>` instead of being inline.
- when no responsive classes are used on non email components, then responsive styles are not applied.
```tsx
   <div className="w-1.5" />
   <Text className="text-sm sm:text-md" />
